{
  "schema_version": "1.3.0",
  "id": "GHSA-47fc-vmwq-366v",
  "modified": "2022-12-13T14:24:18Z",
  "published": "2022-11-26T03:30:27Z",
  "aliases": [
    "CVE-2022-45907"
  ],
  "summary": "PyTorch vulnerable to arbitrary code execution",
  "details": "In PyTorch before trunk/89695, torch.jit.annotations.parse_type_line can cause arbitrary code execution because eval is used unsafely. The fix for this issue is planned to be released in version 1.13.1, there is a release checker in issue #89855.",
  "severity": [
    {
      "type": "CVSS_V3",
      "score": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H"
    }
  ],
  "affected": [
    {
      "package": {
        "ecosystem": "PyPI",
        "name": "torch"
      },
      "ranges": [
        {
          "type": "ECOSYSTEM",
          "events": [
            {
              "introduced": "0"
            },
            {
              "last_affected": "1.13.0"
            }
          ]
        }
      ]
    }
  ],
  "references": [
    {
      "type": "ADVISORY",
      "url": "https://nvd.nist.gov/vuln/detail/CVE-2022-45907"
    },
    {
      "type": "WEB",
      "url": "https://github.com/pytorch/pytorch/issues/88868"
    },
    {
      "type": "WEB",
      "url": "https://github.com/pytorch/pytorch/issues/89855"
    },
    {
      "type": "WEB",
      "url": "https://github.com/pytorch/pytorch/pull/89189"
    },
    {
      "type": "WEB",
      "url": "https://github.com/pytorch/pytorch/commit/767f6aa49fe20a2766b9843d01e3b7f7793df6a3"
    },
    {
      "type": "PACKAGE",
      "url": "https://github.com/pytorch/pytorch"
    }
  ],
  "database_specific": {
    "cwe_ids": [
      "CWE-77"
    ],
    "severity": "CRITICAL",
    "github_reviewed": true
  }
}